### ISSUE TITLE
Bug: Some text formats use CSS instead of semantic HTML

### ISSUE THREAD

**Issue by geryogam** [facebook/lexical#5310]
Lexical version: v0.12.4

## Steps to reproduce

1. Go to https://playground.lexical.dev/.
2. Type the text ‘abc’.
3. Select the text ‘abc’.
4. Click on the ‘Bold’, ‘Italic’, and ‘Underline’ buttons.

## The current behavior

You get this HTML:

```html
<strong class="PlaygroundEditorTheme__textBold PlaygroundEditorTheme__textItalic PlaygroundEditorTheme__textUnderline" data-lexical-text="true">abc</strong>
```

## The expected behavior

You should get this HTML:

```html
<strong data-lexical-text="true"><em><u>abc</u></em></strong>
```

## Analysis

The [WHATWG HTML specification](https://html.spec.whatwg.org/) does not forbid nesting `strong`, `em`, `b`, `i`, `u`, and `s` elements.

[GitHub](https://github.com/) nests those elements for instance (the very same editor I am using to open this issue). Post a comment, type the text ‘abc’, select it, click on the ‘Bold’ and ‘Italic’ buttons, and you get this HTML:

```html
<strong><em>abc</em></strong>
```

[Gmail](https://www.google.com/intl/fr/gmail/about/) nests those elements as well. Compose an email, type the text ‘abc’, select it, click on the ‘Bold’, ‘Italic’, and ‘Underline’ buttons, and you get this HTML:

```html
<b><i><u>abc</u></i></b>
```

[Stack Overflow](https://stackoverflow.com/) nests those elements as well. Ask a question, type the text ‘abc’, select it, click on the ‘Bold’ and ‘Italic’ buttons, and you get this HTML:

```html
<em><strong>abc</strong></em>
```

[Wikipedia](https://www.wikipedia.org/) nests those elements as well. Edit an article, type the text ‘abc’, select it, click on the ‘Bold’ and ‘Italic’ buttons, and you get this HTML:

```html
<i><b>abc</b></i>
```

In other words, the behavior of Lexical which uses CSS instead of semantic HTML to format text in bold, italic, underline, and line-through seems quite unique. It is also inconsistent with how Lexical applies other formats which uses semantic HTML instead of CSS with generic `div` and `span` elements: `p` elements for paragraphs, `h1` to `h6` elements for headings, `blockquote` elements for block quotes, `ul` elements for unordered lists, `ol` elements for ordered lists, `li` elements for list items, `sub` elements for subscripts, `sup` elements for superscripts, `code` elements for code.

**Comment by acywatson** [comment]
I don't necessarily disagree, but I struggle to understand how this is a bug. What are you trying to achieve?

**Comment by geryogam** [comment]
I agree, it's more a feature request than a bug. Should I open a feature request instead?

**Comment by acywatson** [comment]
Yea I think you're both right about this, actually.

We have to be careful to avoid regressing on the accessibility nuances, since those can have a very large blast radius, but I think this one is due for a review. We'll discuss this internally and circle back.
